state_machine/applications/system-check/tests/imu_test.py

```diff
@@ -0,0 +1,256 @@
+    gyro_average = gyro_total / 10
+    mag_change = norm(end_mag - start_mag)
+
+    # return average acc, average gyro, change in mag
+    print("Data Collection Complete")
+    return acc_average, gyro_average, mag_change
+
+
+def stationary_imu_test(result_dict):
+    """
+    Check that norm of gyro is near 0
+    """
+    prompt = "Please leave the cubesat stationary on a table."
+    res = request_imu_data(prompt)
+
+    if res is None:
+        result_dict["IMU_GyroStationary"] = (
+            "Stationary test not completed.", None)
+        return result_dict
+
+    _, gyro, _ = res
+    gyro_string = f"Gyro: {tuple(gyro)} (deg/s)"
+    gyro_stationary = norm(gyro) < 1
+
+    if gyro_stationary:
+        print("Stationary gyroscope reading is near 0 deg/s.")
+    else:
+        print(f"Gyro reading is too large with norm {norm(gyro)} ≥ 1 deg/s.")
+
+    result_dict["IMU_GyroStationary"] = (gyro_string, gyro_stationary)
+    return result_dict
+
+
+def check_gravity_acc(acc_readings):
+    """
+    Given 3 sets of acc readings, check that each set has one
+    correct reading for g, in any order
+    """
+
+    # note if x, y, z readings are correct for each acc value, add to
+    # readings_correct_array
+    readings_correct_array = []
+    for acc in acc_readings:
+        xdir = abs(abs(numpy.dot(acc, numpy.array([1, 0, 0]))) - 9.8) < 1
+        ydir = abs(abs(numpy.dot(acc, numpy.array([0, 1, 0]))) - 9.8) < 1
+        zdir = abs(abs(numpy.dot(acc, numpy.array([0, 0, 1]))) - 9.8) < 1
+        readings_correct_array.append([xdir, ydir, zdir])
+
+    # if there is a True value in each subarray, find its index
+    # (x = 0, y = 1, z = 2)
+    correct_directions = []
+    for i in range(3):
+        correct_index = -1
+        if True in readings_correct_array[i]:
+            correct_index = readings_correct_array[i].index(True)
+        correct_directions.append(correct_index)
+
+    # populate xyz_correct and return
+    # if a direction's index is in correct_directions, we read g correct in
+    # that direction at least once
+    xyz_correct = [(i in correct_directions) for i in range(3)]
+    result = xyz_correct == [True, True, True]
+    return result, xyz_correct
+
+
+def gravity_imu_test(result_dict):
+    """
+    Check that we read g correctly in all 3 directions, given any user
+    input order
+    """
+    # get readings from all 3 sides
+    acc_readings = []
+    for i in range(3):
+        prompt = "Please leave the cubesat flat on one side."
+        if i > 0:
+            prompt = "Please leave the cubesat flat on another side."
+        res = request_imu_data(prompt)
+        if res is None:
+            result_dict["IMU_AccGravity"] = (
+                "Gravity test not completed.", None)
+            return result_dict
+        acc, _, _ = res
+        acc_readings.append(numpy.array(acc))
+
+    result, xyz_correct = check_gravity_acc(acc_readings)
+    xdir, ydir, zdir = xyz_correct
+
+    if result:
```

Comment:
You can just set the result string with the if/else, not have duplication in the 
```python
print(result_string)
result_dict["IMU_AccGravity"] = (result_string, result)
```